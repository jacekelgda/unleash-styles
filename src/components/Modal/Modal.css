.Modal {
  position: fixed;
  width: calc(100% - 100px);
  top: 20%;
  left: 20px;
  transition: .3s opacity ease-out;
  background-color: #fff;
  box-shadow: 0 0 30px rgba(200,200,200,.75);
  padding: 30px;
  z-index: 1002;
}

@media screen and (min-width: 460px) {
  .Modal {
    width: 400px;
    left: calc(50% - 230px);
  }
}

.Modal.isHidden {
  visibility: hidden;
  opacity: 0;
}

.Modal_overlay {
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background-color: rgba(100,100,100,.2);
  transition: .3s opacity ease-out, .3s background-color ease-out;
  z-index: 1001;
}

.Modal_overlay.isHidden {
  visibility: hidden;
  opacity: 0;
  background-color: transparent;
  width: 0;
  height: 0;
}

.Modal_title {
  font: 600 22px/1 'Montserrat',sans-serif;
  text-transform: uppercase;
  text-align: center;
  color: #34323b;
  margin: 0 auto 20px;
}
