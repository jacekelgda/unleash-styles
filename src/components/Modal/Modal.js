/** @flow */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import styles from './Modal.css'

/**
 * Modal component
 */
class Modal extends Component {
  state: {
    isShowing: boolean
  }

  constructor (props: Object) {
    super(props)
    this.state = {
      isShowing: props.isOpen
    }
  }

  hideModal () {
    this.setState({
      isShowing: !this.state.isShowing
    })
  }

  render () {
    const modalClasses = classNames(
      styles.Modal,
      (!this.state.isShowing) && styles.isHidden
    )
    const overlayClasses = classNames(
      styles.Modal_overlay,
      (!this.state.isShowing) && styles.isHidden
    )

    return (
      <div>
        <div className={overlayClasses} onClick={() => this.hideModal()} />
        <div className={modalClasses}>
          <div className={styles.Modal_content}>
            <p className={styles.Modal_title}>{this.props.header}</p>
            {this.props.children}
          </div>
        </div>
      </div>
    )
  }
}

Modal.propTypes = {
  /** Title of modal */
  header: PropTypes.string,
  /** Set true to show modal */
  isOpen: PropTypes.bool,
  /** Components showing in modal */
  children: PropTypes.node.isRequired
}

export default Modal
