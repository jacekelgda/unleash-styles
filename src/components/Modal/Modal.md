Modal component:

    <ModalTrigger label="Open modal">
      <Modal isOpen={false} header={"Modal Test"}>
        <div>
          <p>This is just for testing purposes...</p>
        </div>
      </Modal>
    </ModalTrigger>
