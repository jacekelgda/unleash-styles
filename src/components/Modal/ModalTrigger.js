/** @flow */
import React, { Component } from 'react'
import PropTypes from 'prop-types'

/**
 * Trigger opening modal component ( for styleguide only )
 */
class ModalTrigger extends Component {
  state: {
    isShowing: boolean
  }

  constructor (props: Object) {
    super(props)
    this.state = {
      isShowing: false
    }
  }

  renderModal () {
    return React.cloneElement(this.props.children, {
      isOpen: true
    })
  }

  handleClick () {
    this.setState({
      isShowing: !this.state.isShowing
    })
  }

  render () {
    return (
      <div>
        <span onClick={() => this.handleClick()}>{this.props.label}</span>
        {(this.state.isShowing) && this.renderModal()}
      </div>
    )
  }
}

ModalTrigger.propTypes = {
  /** Label on trigger */
  label: PropTypes.string.isRequired,
  /** Modal */
  children: PropTypes.node.isRequired
}

export default ModalTrigger
