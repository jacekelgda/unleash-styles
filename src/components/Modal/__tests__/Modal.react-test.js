/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import Modal from '../Modal'

test('Modal renders correctly', () => {
  const wrapper = shallow(<Modal isOpen header={'Header'}>
    <div>
      <p>This is just for testing purposes...</p>
    </div>
  </Modal>)
  expect(wrapper).toMatchSnapshot()
})
