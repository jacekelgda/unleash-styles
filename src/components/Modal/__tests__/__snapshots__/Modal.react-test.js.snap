// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="Modal_overlay"
        onClick={[Function]}
    />
    <div
        className="Modal"
    >
        <div
            className="Modal_content"
        >
            <p
                className="Modal_title"
            >
                Header
            </p>
            <div>
                <p>
                    This is just for testing purposes...
                </p>
            </div>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="Modal_overlay"
            onClick={[Function]}
      />
      <div
            className="Modal"
      >
            <div
                  className="Modal_content"
            >
                  <p
                        className="Modal_title"
                  >
                        Header
                  </p>
                  <div>
                        <p>
                              This is just for testing purposes...
                        </p>
                  </div>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Modal
        header="Header"
        isOpen={true}
>
        <div>
                <p>
                        This is just for testing purposes...
                </p>
        </div>
</Modal>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Modal {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div>
            <p>
                        This is just for testing purposes...
            </p>
</div>,
          "header": "Header",
          "isOpen": true,
        },
        "refs": Object {},
        "state": Object {
          "isShowing": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="Modal_overlay"
                    onClick={[Function]}
          />
          <div
                    className="Modal"
          >
                    <div
                              className="Modal_content"
                    >
                              <p
                                        className="Modal_title"
                              >
                                        Header
                              </p>
                              <div>
                                        <p>
                                                  This is just for testing purposes...
                                        </p>
                              </div>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="Modal_overlay"
                    onClick={[Function]}
          />
          <div
                    className="Modal"
          >
                    <div
                              className="Modal_content"
                    >
                              <p
                                        className="Modal_title"
                              >
                                        Header
                              </p>
                              <div>
                                        <p>
                                                  This is just for testing purposes...
                                        </p>
                              </div>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Modal
    header="Header"
    isOpen={true}
>
    <div>
        <p>
            This is just for testing purposes...
        </p>
    </div>
</Modal>,
}
`;
